File: SiiParser/RegexMatch.cs
namespace SiiParser
{
    public class RegexMatch
    {
        public string Value { get; }
        public int Index { get; }
        public int Length { get; }

        public int EndIndex { get; }
        public string TokenType { get; }

        public RegexMatch(string value, int index, int length, string tokenTypeIdentifier)
        {
            this.Value = value;
            this.Index = index;
            this.Length = length;
            this.EndIndex = index + length;
            this.TokenType = tokenTypeIdentifier;
        }
    }
}

File: src/classes/regex-match.ts
import {TokenIdentifier} from "../token-types/token-type";

class RegexMatch {
    public value: string;
    public index: number;

    public get length() {
        return this.value.length;
    }

    public get endIndex() {
        return this.index + length;
    }

    public readonly tokenType: string;

    constructor(value: string, index: number, tokenType: TokenIdentifier) {
        this.value = value;
        this.index = index;
        this.tokenType = tokenType;
    }
}

export default RegexMatch;